RecoMET/METFilters/plugins/EcalBadCalibFilter.cc

```diff
@@ -0,0 +1,228 @@
+// -*- C++ -*-
+//
+// Package:    EcalBadCalibFilter
+// Class:      EcalBadCalibFilter
+//
+/**\class EcalBadCalibFilter EcalBadCalibFilter.cc
+ 
+ Description: <one line class summary>
+ Event filtering to remove events with anomalous energy intercalibrations in specific ECAL channels
+*/
+//
+// Original Authors:  D. Petyt
+//
+ 
+ 
+// include files
+ 
+#include <iostream>
+ 
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+ 
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+ 
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+ 
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "RecoCaloTools/Navigation/interface/CaloNavigator.h"
+ 
+#include "TVector3.h"
+ 
+using namespace std;
+ 
+ 
+class EcalBadCalibFilter : public edm::global::EDFilter<> {
+ 
+public:
+ 
+  explicit EcalBadCalibFilter(const edm::ParameterSet & iConfig);
+  ~EcalBadCalibFilter() override {}
+ 
+private:
+ 
+  // main filter function
+ 
+ bool filter(edm::StreamID, edm::Event & iEvent, const edm::EventSetup & iSetup) const override; 
+ 
+ 
+  // input parameters
+ 
+  // eb rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  ebRHSrcToken_;  
+  // ee rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  eeRHSrcToken_;
+
+  //config parameters (defining the cuts on the bad SCs)
+  const double EBmin_;              // EB rechit et threshold
+  const double EEmin_;              // EE rechit et threshold
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.
This style also applies to python parameter names